catalog/src/main/java/org/killbill/billing/catalog/DefaultTier.java

```diff
@@ -62,6 +67,31 @@ public DefaultTier() {
         blocks = new DefaultTieredBlock[0];
     }
 
+    public DefaultTier(Tier in, TierPriceOverride override, Currency currency) {
+        this.limits = (DefaultLimit[])in.getLimits();
+        this.blocks = new DefaultTieredBlock[in.getTieredBlocks().length];
+
+        for (int i = 0; i < in.getTieredBlocks().length; i++) {
+            if(override != null && override.getTieredBlockPriceOverrides() != null) {
+                final TieredBlock curTieredBlock = in.getTieredBlocks()[i];
+                final TieredBlockPriceOverride overriddenTierBlock = Iterables.tryFind(override.getTieredBlockPriceOverrides(), new Predicate<TieredBlockPriceOverride>() {
+                    @Override
+                    public boolean apply(final TieredBlockPriceOverride input) {
+                        return (input != null && input.getUnitName().equals(curTieredBlock.getUnit().getName()) &&
+                                Double.compare(input.getSize(), curTieredBlock.getSize()) == 0 &&
+                                Double.compare(input.getMax(), curTieredBlock.getMax()) == 0);
+                    }
+
+                }).orNull();
+                blocks[i] = (overriddenTierBlock != null) ? new DefaultTieredBlock(in.getTieredBlocks()[i], overriddenTierBlock, currency) :
```

Comment:
Validating using tiered block's name size and max values to see if any override exists for the current tier block. Same thought as the usage override. Will this not work?